02_housing/2a_affordable_available_place.qmd

```diff
@@ -0,0 +1,251 @@
+    share_affordable_available_50_ami_owner = (at_rent50_owner+Affordable50AMI_owner_vacant)/Below50AMI, 
+    
+    # number affordable and available at 80 AMI per 100 households
+    share_affordable_available_80_ami_all = (at_rent80_all+Affordable80AMI_all_vacant)/Below80AMI, 
+    share_affordable_available_80_ami_renter = (at_rent80_renter+Affordable80AMI_renter_vacant)/Below80AMI, 
+    share_affordable_available_80_ami_owner = (at_rent80_owner+Affordable80AMI_owner_vacant)/Below80AMI
+  )
+```
+
+## (5) Clean and export
+
+### (5a) subgroup file
+
+```{r}
+# combine with overall share_affordable file 
+affordable_2022_subgroup <- read_csv(here::here("02_housing/data/housing_2022_subgroups_city.csv")) %>% 
+  select(state, place, matches("quality"))
+
+# turn long for subgroup output
+available_2022_subgroup <- available_2022_final %>%
+  select(state, place, starts_with("share_affordable_available")) %>% 
+  # create year variable
+  mutate(year = 2022) %>% 
+  # seperate share_afforadable by AMI and the subgroup
+  pivot_longer(cols = c(contains("share_affordable_available")), 
+               names_to = c("available", "subgroup"),
+               names_pattern = "(.+?(?=_[^_]+$))(_[^_]+$)", # this creates two columns - "share_affordable_XXAMI" and "_owner/_renter/_all"
+               values_to = "value") %>% 
+  # pivot_wider again so that each share_affordable by AMI is it's own column with subgroups as rows
+  pivot_wider(
+    names_from = available, 
+    values_from = value
+  ) %>% 
+  # clean subgroup names and add subgroup type column 
+  # remove leading underscore and capitalize words
+  mutate(subgroup = str_remove(subgroup, "_") %>% str_to_title(),
+         subgroup_type = "tenure") %>% 
+  # join with affordable values for final file
+  left_join(affordable_2022_subgroup) %>% 
+  # create a share affordable available quality flag
+  # THIS IS THE SAME AS THE SHARE AFFORDABLE QUALITY FLAG
+  # BECAUSE BOTH ARE BASED ON THE UNDERLYING HOUSEHOLD SAMPLE SIZE AND PUMA CROSSWALK
+  mutate(across(matches("share_.*quality$"),
+                list(share_affordable_available = ~.), 
+                .names = "share_affordable_available_{str_remove(.col, 'share_affordable_')}"), 
+         # subpress counties with too small of sample size
+         across(matches("share_affordable_available.*ami$"), \(x) if_else(is.na(get(cur_column() %>% paste0("_quality"))), NA, x))) %>% 
+  # arrange columns and drop unneeded ones
+  select(year, state, place, subgroup_type, subgroup, matches("available_30"),  matches("available_50"), matches("available_80"))
+
+# export our file as a .csv
+write_csv(available_2022_subgroup, here::here("02_housing/data/available_2022_city.csv"))  
+```
+
+## (5b) overall file
+
+```{r}
+# combine with share_affordable file 
+affordable_2022_overall <- read_csv(here::here("02_housing/data/housing_2022_city.csv"))
+
+# keep what we need
```

Comment:
I don't understand the purpose of this chunk. Since we already have the affordable metrics output in another file, I don't know what the purpose of merging them back on here is? We also now have 3 rows per place even though they are all supposed to be in the subgroup "all"?

Reply:
Same as above, using to add the data quality flag.

Reply:
OK I think now that the previous issue is resolved this one is too.